x/erc20/keeper/genesis.go

```diff
@@ -0,0 +1,121 @@
+		}
+		stateDB := statedb.New(ctx, k.evmKeeper, statedb.TxConfig{})
+		code, err := k.generateContractCode(ctx, stateDB, coinMeta, contractAddr)
+		if err != nil {
+			return fmt.Errorf("error while getting contract code for genesis pair denom %s. %w", pair.Denom, err)
+		}
+		stateDB.SetCode(contractAddr, code)
+
+		if err := stateDB.Commit(); err != nil {
+			return fmt.Errorf("error while storing contract for genesis pair denom %s. %w", pair.Denom, err)
+		}
+
+		id := pair.GetID()
+		k.SetTokenPair(ctx, pair)
+		k.SetDenomMap(ctx, pair.Denom, id)
+		k.SetERC20Map(ctx, pair.GetERC20Contract(), id)
+	}
+
+	return nil
+}
+
+// getGenesisTokenPairMeta is a helper function to generate token pair metadata for the genesis token pairs
+func (k Keeper) getGenesisTokenPairMeta(ctx sdk.Context, pair types.TokenPair, index int) (banktypes.Metadata, error) {
+	displayName := fmt.Sprintf("genToken%d", index)
+	meta := banktypes.Metadata{
+		Description: fmt.Sprintf("Genesis token pair number %d", index),
+		DenomUnits: []*banktypes.DenomUnit{
+			{Denom: pair.Denom, Exponent: 0, Aliases: []string{fmt.Sprintf("uGenToken%d", index)}},
+			{Denom: displayName, Exponent: 6},
+		},
+		Base:    pair.Denom,
+		Display: displayName,
+		Name:    displayName,
+		Symbol:  displayName,
+	}
+	if err := k.verifyMetadata(ctx, meta); err != nil {
+		return banktypes.Metadata{}, errorsmod.Wrapf(
+			types.ErrInternalTokenPair, "coin metadata is invalid for genesis pair denom %s", pair.Denom,
+		)
+	}
+	return meta, nil
+}
+
+// generateContractCode is a helper function to generate
+// the ERC20 contract code to be stored at genesis when token pairs are provided
+func (k Keeper) generateContractCode(ctx sdk.Context, stateDB *statedb.StateDB, coinMeta banktypes.Metadata, contractAddr common.Address) ([]byte, error) {
+	evm := k.newEVM(ctx, stateDB)
+	data, err := getContractDataBz(coinMeta)
+	if err != nil {
+		return nil, err
+	}
+	// Initialize a new contract and set the code that is to be used by the EVM.
+	// The contract is a scoped environment for this execution context only.
+	sender := vm.AccountRef(types.ModuleAddress)
+	contract := vm.NewContract(sender, vm.AccountRef(contractAddr), big.NewInt(0), 1000000)
+	contract.Code = data
+
+	return evm.Interpreter().Run(contract, nil, false)
+}
+
+// getEVMConfig is a helper function to get an EVM config
+// needed to instantiate the EVM when token pairs are provided
+// at genesis
+func (k Keeper) getEVMConfig(ctx sdk.Context) *statedb.EVMConfig {
+	params := k.evmKeeper.GetParams(ctx)
+	ethCfg := params.ChainConfig.EthereumConfig(k.evmKeeper.ChainID())
+
+	baseFee := k.evmKeeper.GetBaseFee(ctx, ethCfg)
+	return &statedb.EVMConfig{
+		Params:      params,
+		ChainConfig: ethCfg,
+		CoinBase:    common.Address{},
+		BaseFee:     baseFee,
+	}
+}
+
+// newEVM is a helper function used during genesis
+// to instantiate the EVM to set genesis state (token pairs)
```

Comment:
nit: the function doesn't know nothing about token pairs
```suggestion
// to instantiate the EVM to set genesis state
```